src/state/lottery/fetchUnclaimedUserRewards.ts

```diff
@@ -0,0 +1,209 @@
+import BigNumber from 'bignumber.js'
+import { ethers } from 'ethers'
+import { LotteryStatus, LotteryTicket, LotteryTicketClaimData } from 'config/constants/types'
+import { LotteryUserGraphEntity, LotteryRoundGraphEntity, UserRound } from 'state/types'
+import { multicallv2 } from 'utils/multicall'
+import lotteryV2Abi from 'config/abi/lotteryV2.json'
+import { getLotteryV2Address } from 'utils/addressHelpers'
+import { BIG_ZERO } from 'utils/bigNumber'
+import {
+  getViewUserTicketInfoCalls,
+  mergeViewUserTicketInfoMulticallResponse,
+  processRawTicketsResponse,
+} from './helpers'
+
+interface RoundDataAndUserTickets {
+  roundId: string
+  userTickets: LotteryTicket[]
+  finalNumber: string
+}
+
+const lotteryAddress = getLotteryV2Address()
+
+const fetchCakeRewardsForTickets = async (
+  winningTickets: LotteryTicket[],
+): Promise<{ ticketsWithUnclaimedRewards: LotteryTicket[]; cakeTotal: BigNumber }> => {
+  const calls = winningTickets.map((winningTicket) => {
+    const { roundId, id, rewardBracket } = winningTicket
+    return {
+      name: 'viewRewardsForTicketId',
+      address: lotteryAddress,
+      params: [roundId, id, rewardBracket],
+    }
+  })
+  const cakeRewards = await multicallv2(lotteryV2Abi, calls)
+
+  const cakeTotal = cakeRewards.reduce((accum: BigNumber, cakeReward: ethers.BigNumber[]) => {
+    return accum.plus(new BigNumber(cakeReward[0].toString()))
+  }, BIG_ZERO)
+
+  const ticketsWithUnclaimedRewards = winningTickets.map((winningTicket, index) => {
+    return { ...winningTicket, cakeReward: cakeRewards[index] }
+  })
+  // TODO: Remove log. To help in testing when verifying winning tickets.
+  console.log('winning tickets: ', ticketsWithUnclaimedRewards)
+  return { ticketsWithUnclaimedRewards, cakeTotal }
+}
+
+const getRewardBracketByNumber = (ticketNumber: string, finalNumber: string): number => {
+  // Winning numbers are evaluated right-to-left in the smart contract, so we reverse their order for validation here:
+  // i.e. '1123456' should be evaluated as '6543211'
+  const ticketNumAsArray = ticketNumber.split('').reverse()
+  const winningNumsAsArray = finalNumber.split('').reverse()
+  const matchingNumbers = []
+
+  // The number at index 6 in all tickets is 1 and will always match, so finish at index 5
+  for (let index = 0; index < winningNumsAsArray.length - 1; index++) {
+    if (ticketNumAsArray[index] !== winningNumsAsArray[index]) {
+      break
+    }
+    matchingNumbers.push(ticketNumAsArray[index])
+  }
+
+  // Reward brackets refer to indexes, 0 = 1 match, 5 = 6 matches. Deduct 1 from matchingNumbers' length to get the reward bracket
+  const rewardBracket = matchingNumbers.length - 1
```

Comment:
Ignore for now: seems like a good candidate for Enum instead of array with indices. Not worth spending time refactoring this now though.

Reply:
Good point. They're used in a few different places - this could be a neat improvement